cpp/src/structure/graph_view_impl.cuh

```diff
@@ -883,6 +883,46 @@ weight_t graph_view_t<
   return ret;
 }
 
+template <typename vertex_t,
+          typename edge_t,
+          typename weight_t,
+          bool store_transposed,
+          bool multi_gpu>
+weight_t
+graph_view_t<vertex_t, edge_t, weight_t, store_transposed, multi_gpu, std::enable_if_t<multi_gpu>>::
+  compute_total_edge_weight(raft::handle_t const& handle) const
+{
+  return transform_reduce_e(
+    handle,
+    *this,
+    edge_src_dummy_property_t{}.view(),
+    edge_dst_dummy_property_t{}.view(),
+    [] __device__(auto, auto, weight_t wt, auto, auto) { return wt; },
+    weight_t{0});
+}
+
+template <typename vertex_t,
+          typename edge_t,
+          typename weight_t,
+          bool store_transposed,
+          bool multi_gpu>
+weight_t graph_view_t<vertex_t,
+                      edge_t,
+                      weight_t,
+                      store_transposed,
+                      multi_gpu,
+                      std::enable_if_t<!multi_gpu>>::compute_total_edge_weight(raft::handle_t const&
+                                                                                 handle) const
+{
+  return transform_reduce_e(
+    handle,
+    *this,
+    edge_src_dummy_property_t{}.view(),
+    edge_dst_dummy_property_t{}.view(),
+    [] __device__(auto, auto, weight_t wt, auto, auto) { return wt; },
+    weight_t{0});
+}
+
```

Comment:
Can we use common functor both for mg sg ompute_total_edge_weight?

Reply:
I'll try making a shared detail method.

Reply:
Done in next push